File: src/main/java/net/softbell/bsh/domain/converter/ItemTypeRuleConverter.java
package net.softbell.bsh.domain.converter;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

import net.softbell.bsh.domain.ItemTypeRule;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : IoT 아이템 타입 자료형 DB 컨버터
 */
@Converter(autoApply = true)
public class ItemTypeRuleConverter implements AttributeConverter<ItemTypeRule, Integer>
{
	@Override
	public Integer convertToDatabaseColumn(ItemTypeRule pinTypeRule)
	{
		if (pinTypeRule == null)
			return null;
		
		return pinTypeRule.getCode();
	}

	@Override
	public ItemTypeRule convertToEntityAttribute(Integer code)
	{
		return ItemTypeRule.ofLegacyCode(code);
	}
}

File: src/main/kotlin/net/softbell/bsh/domain/converter/ItemTypeRuleConverter.kt
package net.softbell.bsh.domain.converter

import net.softbell.bsh.domain.ItemTypeRule
import javax.persistence.AttributeConverter
import javax.persistence.Converter

/**
 * @author : Bell(bell@softbell.net)
 * @description : IoT 아이템 타입 자료형 DB 컨버터
 */
@Converter(autoApply = true)
class ItemTypeRuleConverter : AttributeConverter<ItemTypeRule, Int> {
    override fun convertToDatabaseColumn(pinTypeRule: ItemTypeRule): Int {
        return pinTypeRule.code
    }

    override fun convertToEntityAttribute(code: Int): ItemTypeRule? {
        return ItemTypeRule.ofLegacyCode(code)
    }
}